docs/standard/data/xml/extending-the-dom.md

```diff
@@ -10,237 +10,243 @@ author: "mairaw"
+
+There are two general guidelines when deriving new classes:
+
+- It is recommended that you never derive from the <xref:System.Xml.XmlNode> class. Instead, it is recommended that you derive classes from the class corresponding to the node type that you are interested in. For example, if you want to return additional information on attribute nodes, you can derive from the <xref:System.Xml.XmlAttribute> class.
+
+- Except for the node creation methods, it is recommended that when overriding a function, you should always call the base version of the function and then add any additional processing.
+
+## Creating Your Own Node Instances
+
+The <xref:System.Xml.XmlDocument> class contains node creation methods. When an XML file is loaded, these methods are called to create the nodes. You can override these methods so that your node instances are created when a document is loaded. For example, if you have extended the <xref:System.Xml.XmlElement> class, you would inherit the <xref:System.Xml.XmlDocument> class and override the <xref:System.Xml.XmlDocument.CreateElement%2A> method.
+
+The following example shows how to override the <xref:System.Xml.XmlDocument.CreateElement%2A> method to return your implementation of the <xref:System.Xml.XmlElement> class.
+
+```vb
+Class LineInfoDocument
+    Inherits XmlDocument
+        Public Overrides Function CreateElement(prefix As String, localname As String, nsURI As String) As XmlElement
+        Dim elem As New LineInfoElement(prefix, localname, nsURI, Me)
+        Return elem
+    End Function 'CreateElement
+End Class 'LineInfoDocument
+```
+
+```csharp
+class LineInfoDocument : XmlDocument {
+  public override XmlElement CreateElement(string prefix, string localname, string nsURI) {
+  LineInfoElement elem = new LineInfoElement(prefix, localname, nsURI, this);
+  return elem;
+  }
+```
+
+## Extending a Class
+
+To extend a class, derive your class from one of the existing DOM classes. You can then override any of the virtual methods or properties in the base class, or add your own.
+
+In the following example, a new class is created, which implements the <xref:System.Xml.XmlElement> class and the <xref:System.Xml.IXmlLineInfo> interface. Additional methods and properties are defined which allows users to gather line information.
+
+```vb
+Class LineInfoElement
+   Inherits XmlElement
+   Implements IXmlLineInfo
+   Private lineNumber As Integer = 0
+   Private linePosition As Integer = 0
+
+   Friend Sub New(prefix As String, localname As String, nsURI As String, doc As XmlDocument)
+      MyBase.New(prefix, localname, nsURI, doc)
+      CType(doc, LineInfoDocument).IncrementElementCount()
+   End Sub 'New
```

Comment:
Note the highlighted characters here. It usually indicates some unusual (non-printing?) character in the file. Here, each of those comments can be removed -- they were auto-generated by the C#-to-VB conversion tool.